filters thread safety /7 (#22296)
dotnet/AspNetCore.Docs: aspnetcore/mvc/controllers/filters.md

Middleware filters run at the same stage of the filter pipeline as Resource filters, before model binding and after the rest of the pipeline.

## Thread safety

> When calling `Add` on a singleton with a new instance of the filter, instead if a type, the filter is not thread-safe.

Reviewer: ```suggestion
When passing an *instance* of a filter into `Add`, instead of its `Type`, the filter is a singleton and is **not** thread-safe.
```

Middleware filters run at the same stage of the filter pipeline as Resource filters, before model binding and after the rest of the pipeline.

## Thread safety

> When calling `Add` on a singleton with a new instance of the filter, instead if a type, the filter is not thread-safe.

Reviewer: ```suggestion
When passing an *instance* of a filter into `Add`, instead of its `Type`, the filter is a singleton and is **not** thread-safe.
```
